### ISSUE TITLE
Custom script icons render too small on Retina display

### ISSUE THREAD

**Issue by samuelfine** [godotengine/godot#96069]
### Tested versions

v4.3.stable.official [77dcf97d8]

### System information

Godot v4.3.stable - macOS 14.6.1 - GLES3 (Compatibility) - Intel(R) UHD Graphics 617 - Intel(R) Core(TM) i5-8210Y CPU @ 1.60GHz (4 Threads)

### Issue description

When setting a custom `@icon` on a script, the icon renders smaller than expected in the Scene Tree. The icon preview in the FileSystem renders at the correct size.

<img width="272" alt="icon-issue" src="https://github.com/user-attachments/assets/93381422-d34d-47d7-b908-29ee2f83f9b9">

### Steps to reproduce

- Create a new script
- Add a custom `@icon` pointed at an asset in the FileSystem (.png and .svg both tested)

```
@icon ("res://assets/icons/icon_save.svg")
extends Node2D
class_name SaveDataResource
```

- Observe icon in FileSystem pane at incorrect size

### Minimal reproduction project (MRP)

[customiconsizeissue.zip](https://github.com/user-attachments/files/16740466/customiconsizeissue.zip)

**Comment by detomon** [comment]
Checking "Scale With Editor Scale" in the import tab and then reimporting it should resize the icon to the correct scale.

![import](https://github.com/user-attachments/assets/b724d58e-7918-4864-a101-78641dc07f8c)

**Comment by samuelfine** [comment]
@detomon Good catch, this does fix the problem for SVG icons! Does lead me to a few additional questions though:

- Should this be checked/true by default? I can't think of any reason why someone would want to use an SVG in the Editor but _not_ scale it to match.
- Should **only** SVG assets be allowed as custom `@icon`s? Looks like all the [built-in icons](https://github.com/godotengine/godot/tree/e63c40e59c6650bdba02c2ceff3390ee9273ac46/editor/icons) are SVG and I don't see that `Scale With Editor Scale` option for importing PNG assets, which leads me to believe the Editor prefers SVG for `@icon`. And I wouldn't be against the idea of requiring SVG for custom icons, for the sake of simplicity and ensuring accurate rendering.

The [documentation](https://docs.godotengine.org/en/stable/classes/class_@gdscript.html#class-gdscript-annotation-icon) doesn't specify a filetype, so at the very least it feels like there's an opportunity for providing clearer user guidance.
